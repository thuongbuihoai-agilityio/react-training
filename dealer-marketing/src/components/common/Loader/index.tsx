import React, { memo } from "react";
import styleLoader from "./loader.module.css";

const LoadingPage: React.FC = () => <div className={styleLoader.loader} />;

export default memo(LoadingPage);
