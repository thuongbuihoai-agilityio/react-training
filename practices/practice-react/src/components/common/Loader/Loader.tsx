import React, { memo } from "react";
import "./loader.css";

const Loader: React.FC = memo(() => {
  return <div data-testid="loading-page" className="loader" />;
});

export default Loader;
