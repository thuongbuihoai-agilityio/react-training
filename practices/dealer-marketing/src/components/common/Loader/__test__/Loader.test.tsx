import { render } from "@testing-library/react";
import Loader from "../index";

describe("Loader component", () => {
  test("Should render Loader component", () => {
    const { getByTestId } = render(<Loader />);
    const loader = getByTestId("loader");
    expect(loader).toBeInTheDocument();
  });

  test("Matches snapshot", () => {
    const { container } = render(<Loader />);
    expect(container).toMatchSnapshot();
  });
});
