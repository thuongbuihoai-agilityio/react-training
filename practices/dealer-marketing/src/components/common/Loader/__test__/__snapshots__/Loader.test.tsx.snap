// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component Matches snapshot 1`] = `
<div>
  <div
    data-testid="loader"
  />
</div>
`;
